packages/rewriter/package.json

```diff
@@ -15,7 +15,7 @@
     "chai": "4.2.0",
     "chai-as-promised": "7.1.1",
     "debug": "4.3.2",
```

Comment:
Do we want to make other dependencies optimistic to avoid duplicate dependencies between hoisted packages in the future, like you've done for lodash? If we're worried about the risk of pulling in minor versions of packages we could lock loadash to the highest common version(looks like `4.17.21`) and not make it optimistic to be consistent about how we're handling dependencies.

```
    "debug": "^4.3.2",
```

Reply:
Yeah, I had actually done that originally in the PR, but then reverted in https://github.com/cypress-io/cypress/pull/17285/commits/a32e015b3174bdd5c87de93bf61b3a2f1ff3a3c3 as I was trying to isolate an issue around code-signing which ended up being unrelated.

Looking at the build contents in [GrandPerspective](http://grandperspectiv.sourceforge.net/), I noticed Lodash was the most obvious duplicated dependency - but I agree we should default to using `^` wherever possible, so that the pinned dependencies stand out. For instance, `proxy-as-env` needs to be pinned to `1.0.0` as `1.1.0` breaks things.